src/pages/task/actionButtons/reducer.ts

```diff
@@ -18,15 +18,18 @@ interface State {
   hasFetchedLastExecuted: boolean;
```

Comment:
Sorry i hadn't noticed this before this should be in a folder on its own. 
The PreviousCommits logic and tests should be in the same folder. They can live as a  sub directory in actionButtons.